### PR TITLE
411: allow users to create new zoning district label

### LINKED ISSUE #411: Users can't create a new district type
New districts do get created sometimes, and the app doesn't currently allow the user to input a new zoning district label

### PR DESCRIPTION
**Pull request by trbmcginnis**
Using `ember-power-select-with-create`

Includes an action to check whether the input in the search box matches the options list. If there is a match, "Add" does not appear in the dropdown. If there are no matches, "Add" will appear, prompting the user to add their new district. 

Includes an argument `canAddNewDistrict` that we set to false for commercial overlays so that "Add" does not show up at all for this step. 

https://github.com/cibernox/ember-power-select-with-create

Closes #411

### CODE REVIEW

**Comment by trbmcginnis** [comment]
Screen Shot 2019-03-13 at 2 53 53 PM

**Comment by julialucyhogan** [comment]
perhaps a test? :)

**Comment by trbmcginnis** [comment]
> perhaps a test? :)

Will do !

**Comment by hannahkates** [comment]
I think the "Add __...." option should be listed last or only show up if what the user has typed doesn't match anything in the existing list. That would help us avoid situations like the one below where the users default option is the create a poorly capitalized new label for something that already exists

image

**Comment by allthesignals** [comment]
@trbmcginnis this is great, just need a test :)

**Comment by julialucyhogan** [comment]
To @hannahkates's point .. I agree in theory, but definitely don't think that's a hill worth dying on. If it's easy enough to configure the addon then yes let's. But if it's not, then the user can just deal.

**Comment by trbmcginnis** [comment]
> I think the "Add __...." option should be listed last or only show up if what the user has typed doesn't match anything in the existing list. That would help us avoid situations like the one below where the users default option is the create a poorly capitalized new label for something that already exists

Cool, I'm going to look into this and see what I can do

File: yarn.lock [hunk 1]
```diff
@@ -3272,7 +3272,10 @@ cartobox-promises-utility@nycplanning/cartobox-promises-utility#v1.0.2:
   resolved "https://codeload.github.com/nycplanning/cartobox-promises-utility/tar.gz/57f1c96a0bf03c47e048c30d202643cc49db3c69"
   dependencies:
     ember-cli-babel "^6.6.0"
+<<<<<<< HEAD
     ember-fetch "5.0.0"
+=======
+>>>>>>> bfad4be8a1452cb6895bfdf0cae8a109859af731
```
Comment (allthesignals) [inline yarn.lock]: Eeek! Looks like some merge conflict markup got in here

Comment (trbmcginnis) [reply yarn.lock]: Ahh! Will fix

File: app/components/project-geometries/modes/draw/feature-label-form.js [hunk 2]
```diff
@@ -55,6 +55,23 @@ export default class FeatureLabelFormComponent extends Component {
...
+    // array of all elements that match between options array and searchOptions array
+    const matchArray = options.filter(function(item) {
+      return searchOptions.includes(item);
+    });
+    // if matchArray has no elements, return true and show "Add"
+    if (matchArray.length < 1) {
+      return true;
+    }
+    return false;
+  }
```
Comment (allthesignals) [inline app/components/project-geometries/modes/draw/feature-label-form.js]: Why does this return something? Should it be a computed property? Maybe I don't understand `showCreateWhen` enough... but if this works, very interesting...

File: tests/integration/components/project-geometries/modes/draw/feature-label-form-test.js [hunk 3]
```diff
@@ -192,6 +192,7 @@ module('Integration | Component | project-geometries/modes/draw/feature-label-fo
     await click('.ember-power-select-search');
     await fillIn('.ember-power-select-search-input', 'Option 4');
     // there will be only one option available
+    assert.equal(find('.ember-power-select-option').textContent.trim(), 'Add "Option 4"...');
```
Comment (allthesignals) [inline tests/integration/components/project-geometries/modes/draw/feature-label-form-test.js]: 👍

File: app/components/project-geometries/modes/draw/feature-label-form.js (hunk 2 above)
Comment (trbmcginnis) [reply app/components/project-geometries/modes/draw/feature-label-form.js]: I followed the way that they did it here in the docs https://github.com/cibernox/ember-power-select-with-create, where they used an action. "You can provide a callback showCreateWhen, which will be called whenever the user types into the search field. If you return true, the create option will be shown. If you return false, it won't be shown." Let me know if you still think it should be something else

Comment (trbmcginnis) [reply app/components/project-geometries/modes/draw/feature-label-form.js]: I followed the way that they did it here in the docs https://github.com/cibernox/ember-power-select-with-create, where they used an action. "You can provide a callback showCreateWhen, which will be called whenever the user types into the search field. If you return true, the create option will be shown. If you return false, it won't be shown." Let me know if you still think it should be something else

**Pull request merged**